src/components/FileBrowser/FileBrowser.vue

```diff
@@ -0,0 +1,252 @@
+          icon: item._modelType in ICON_MAP ? ICON_MAP[item._modelType] : 'file',
+        }));
+        this.loading = false;
+        return rows;
+      },
+    },
+  },
+  watch: {
+    location (newval, oldval) {
+      // force reset pagination when location changes.
+      if (newval.id !== oldval.id){
+        console.log(newval, oldval);
+        this.pagination.page = 1; 
+      }
+    } 
+  },
+  computed: {
+    folderParams() {
+      return {
+        parentType: this.location.type,
+        parentId: this.location.id,
+        limit: this.pagination.rowsPerPage,
+        offset: (this.pagination.page - 1) * this.pagination.rowsPerPage
+      }
+    },
+    itemParams() {
+      const limit = this.counts.nFolders > this.folderParams.offset
+          ? this.pagination.rowsPerPage - (this.counts.nFolders - this.folderParams.offset)
+          : this.pagination.rowsPerPage;
+      const offset = this.folderParams.offset > 0
+          ? this.folderParams.offset - this.counts.nFolders
+          : 0;
+      if (limit <= 0 || (this.location.type !== 'folder')) {
+        return null; // No items should be shown on the current page.
+      }
+      return { folderId: this.location.id, limit, offset };
+    },
+    login () { return this.girderRest.user ? this.girderRest.user.login : ''; },
+    totalItems () { return this.counts.nFolders + this.counts.nItems; },
+  },
+  methods: {
+    toggleAll() {
+      if (this.selected.length === this.rows.length) { 
+        this.selected = [];
+      } else {
+        this.selected = this.rows.slice();
+      }
+    },
+  },
+  created() {
+    this.girderRest.$on('login', this.$forceUpdate);
+  },
+};
+</script>
+
+<template lang="pug">
+v-data-table.girder-file-browser-component(
+    select-all,
+    :headers-length="4",
+    v-model="selected",
+    :pagination.sync="pagination",
+    :items="rows",
+    :total-items="totalItems",
+    item-key="id",
+    class="elevation-1")
+  
+  //- Header Slot
+  template(slot="headers", slot-scope="props")
+    tr.secondary
+      th(width="1%")
+        v-checkbox.pr-2(
+            color="accent",
+            hide-details,
+            :input-value="props.all",
+            :indeterminate="selected.length > 0 && !props.all",
+            @click.native="toggleAll")
+      th.pl-0(width="80%")
+        v-breadcrumbs.pl-0
+          v-icon.mdi-24px(slot='divider', color='accent') {{ $vuetify.icons.chevron }}
+          v-breadcrumbs-item.selectable(@click.native="$emit('update:location', breadcrumb.root)")
+            v-icon.mdi-24px(color='accent') {{ $vuetify.icons.globe }}
+            | &nbsp; {{ login }}
+          v-breadcrumbs-item(v-for="item in breadcrumb.path", :key="`${item.id}.crumb`", @click.native="$emit('update:location', item)") {{ item.name }}
+      th(width="1%").pr-0
+        v-btn(flat, small, v-if="newFolderEnabled")
+          v-icon.mdi-24px.mr-1(left, color="accent") {{  $vuetify.icons.folderNew }}
+          | NEW FOLDER
```

Comment:
I believe v-btn will automatically capitalize text in buttons. I'd prefer we keep them "Sentence case" in markup. The upcoming iteration of material design seems to remove the idea of uppercasing everything so it is a forward-compatible practice.